busypie/awaiter.py

```diff
@@ -39,11 +47,19 @@ def _raise_exception_if_not_ignored(self, e: Exception):
                 (ignored_exceptions and e.__class__ not in ignored_exceptions):
             raise e
 
-    def _validate_wait_constraint(self, condition_evaluator: ConditionEvaluator, start_time: float):
-        if (time.time() - start_time) > self._condition.wait_time_in_secs:
+    def _validate_condition_based_constraint(self, condition_evaluator: ConditionEvaluator, condition: bool):
```

Comment:
You can extract validate into its own class

Reply:
The term at_least doesn't fit well IMO. how about min_wait_time & max_wait_time
or lower_bound_wait and upper_bound_wait?

Reply:
fixed